### PR TITLE
Update go libraries page with new intro

### PR DESCRIPTION
**Pull request by svrnm**
- Makes use of the newly introduced intro shortcode for the go page on instrumentation libraries. 
- Some additional changes to be closer to what other languages (Js, etc.) have on that page.

Preview: https://deploy-preview-3826--opentelemetry.netlify.app/docs/languages/go/libraries/

### CODE REVIEW

File: content/en/docs/languages/go/libraries.md [hunk 1]
```diff
@@ -7,12 +7,18 @@ aliases:
 weight: 40
 ---
 
-Go does not support truly automatic instrumentation like other languages today.
-Instead, you'll need to depend on
+{{% docs/languages/libraries-intro Go %}}
+
+## Use Instrumentation Libraries
```
Comment (chalin) [inline content/en/docs/languages/go/libraries.md:12]:
```suggestion
## Use instrumentation libraries
```

File: content/en/docs/languages/go/libraries.md [hunk 2]
```diff
@@ -7,12 +7,18 @@ aliases:
...
+
+If a library does not come with OpenTelemetry out of the box, you can use
 [instrumentation libraries](/docs/specs/otel/glossary/#instrumentation-library)
-that generate telemetry data for a particular instrumented library. For example,
-the instrumentation library for `net/http` will automatically create spans that
-track inbound and outbound requests once you configure it in your code.
+in order to generate telemetry data for a library or framework.
+
+For example,
+[the instrumentation library for `net/http`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp)
```
Comment (chalin) [inline content/en/docs/languages/go/libraries.md:19]:
```suggestion
For example, the
[instrumentation library for `net/http`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp)
```

File: content/en/docs/languages/go/libraries.md [hunk 3]
```diff
@@ -7,12 +7,18 @@ aliases:
...
+will automatically create [spans](/docs/concepts/signals/traces/#spans) and
+metrics(/docs/concepts/signals/metrics) based on the HTTP requests.
```
Comment (chalin) [inline content/en/docs/languages/go/libraries.md:21]: Did you mean to make a link here?

```suggestion
[metrics](/docs/concepts/signals/metrics/) based on the HTTP requests.
```

File: content/en/docs/languages/go/libraries.md [hunk 4]
```diff
@@ -42,6 +48,7 @@ A full list of instrumentation libraries available can be found in the
 Instrumentation libraries can do things like generate telemetry data for inbound
 and outbound HTTP requests, but they don't instrument your actual application.
 
-To get richer telemetry data, use [manual instrumentation](../instrumentation/)
-to enrich your telemetry data from instrumentation libraries with
-instrumentation from your running application.
+To get richer telemetry data, add your own
+[instrumentation](../instrumentation/) to your code to enrich your telemetry
+data from instrumentation libraries with instrumentation from your running
+application.
```
Comment (chalin) [inline content/en/docs/languages/go/libraries.md:54]: Hmm, how about:

```suggestion
Enrich your telemetry data by integrating [custom instrumentation](../instrumentation/) into your code. This supplements the standard library telemetry, and can offer deeper insights into your running application.
```

**Comment by chalin** [comment]
I made a copyedit pass. PTAL at inline comments.

**Comment by cartermp** [comment]
Formatting failures!

File: content/en/docs/languages/go/libraries.md [hunk 5]
```diff
@@ -7,12 +7,18 @@ aliases:
...
+{{% docs/languages/libraries-intro Go %}}
+
+## Use instrumentation libraries
+
+If a library does not come with OpenTelemetry out of the box, you can use
 [instrumentation libraries](/docs/specs/otel/glossary/#instrumentation-library)
-that generate telemetry data for a particular instrumented library. For example,
-the instrumentation library for `net/http` will automatically create spans that
-track inbound and outbound requests once you configure it in your code.
+in order to generate telemetry data for a library or framework.
```
Comment (theletterf) [inline content/en/docs/languages/go/libraries.md:16]:
```suggestion
to generate telemetry data for a library or framework.
```

Shorter is always better!

File: content/en/docs/languages/go/libraries.md [hunk 6]
```diff
@@ -7,12 +7,18 @@ aliases:
...
+
+For example, the
+[instrumentation library for `net/http`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp)
+will automatically create [spans](/docs/concepts/signals/traces/#spans) and
```
Comment (theletterf) [inline content/en/docs/languages/go/libraries.md:20]:
```suggestion
automatically creates [spans](/docs/concepts/signals/traces/#spans) and
```

**Comment by svrnm** [comment]
@open-telemetry/go-approvers PTAL!

**Review by theletterf — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**